File: AndroidPatternsApp/jmvp-app/src/main/java/com/woowrale/jmvp/ui/search/remote/RemoteSearchView.java
package com.woowrale.jmvp.ui.search.remote;

import com.woowrale.jmvp.data.model.Contact;
import com.woowrale.jmvp.ui.adapters.ContactsAdapter;
import com.woowrale.jmvp.ui.base.BasePresenter;
import io.reactivex.subjects.PublishSubject;

import java.util.List;

public interface RemoteSearchView extends BasePresenter.View {

    ContactsAdapter getmAdapter();

    List<Contact> getContactsList();

    PublishSubject<String> getPublishSubject();

}

File: InstantSearchApp/kapp/src/main/java/com/woowrale/instantsearch/kapp/ui/search/remote/RemoteSearchView.kt
package com.woowrale.instantsearch.kapp.ui.search.remote

import com.woowrale.instantsearch.kapp.data.model.Contact
import com.woowrale.instantsearch.kapp.ui.adapters.ContactsAdapter
import com.woowrale.instantsearch.kapp.ui.base.BasePresenter
import io.reactivex.subjects.PublishSubject

interface RemoteSearchView : BasePresenter.View {

    val publishSubject: PublishSubject<String>

    fun  setContactsList(contacts: List<Contact>)

    fun getmAdapter(): ContactsAdapter

}
